website/src/models/group.js

```diff
@@ -444,19 +444,45 @@ GroupSchema.methods.leave = function(user, keep, mainCb){
         var leader = group.leader;
 
         if(leader == user._id || !~group.members.indexOf(leader)){
-          var seniorMember = _.find(group.members, function (m) {return m != user._id});
 
-          // could not exist in case of public guild with 1 member who is leaving
-          if(seniorMember){
-            if (leader == user._id || !~group.members.indexOf(leader)) {
-              update['$set'] = update['$set'] || {};
-              update['$set'].leader = seniorMember;
+          var seniorMember = undefined;
+          var today = new Date();
+          var lastWeek = new Date(today.getFullYear(), today.getMonth(), today.getDate() - 7);
+
+          User.findOne({
+            '_id': { $in : group.members, $ne : leader },
+            'auth.blocked' : { $exists: false },
+            'flags.chatRevoked': { $exists: false },
+            'lastCron': { $gt: lastWeek }
+          })
+          .sort({'lastCron': -1})
```

Comment:
According to this http://stackoverflow.com/questions/25058007/mongodb-difference-between-orderby-and-sort, `sort` should be avalaible only to `find()` and not `findOne()` but they also mention that it should produce an error, which in this case doesn't seem the case.

Anyway the documentation about `findOne()` says that the document is returned according to natural order http://docs.mongodb.org/manual/reference/method/db.collection.findOne/

The solutions seems to be:

1. Use `find()` with `limit(1)` and `sort()`
2. Use `findOne()` with `$orderBy`